File: Lab 1 Solutions/Program8.py
my_input = [int(x) for x in input("Enter the number list here: ").split()]

def bubble_sort(bad_list):
    sorted = False
    while not sorted:
        sorted = True
        for i in range(len(bad_list)-1):
            if bad_list[i] > bad_list[i+1]:
                sorted = False
                bad_list[i], bad_list[i+1] = bad_list[i+1], bad_list[i]
                
bubble_sort(my_input)
print("The Sorted List is: " + str(my_input))

File: Lab 1 Solutions/Program8.c
#include <stdio.h>

int * bubble_sort(int *a, int num){
	int i, j, temp;
	for(i = 0; i < num; i++){
		for(j = i+1; j < num; j++){
			if(a[i] > a[j]){
				temp = a[i];
				a[i] = a[j];
				a[j] = temp;
			}
		}
	}
	return a;
}

int main(){
	int array[6] = {12, 5, 8, 16, 65, 10}, i, *a = NULL;
	a = bubble_sort(array, 6);
	printf("The sorted array is: \n");
	for(i = 0; i < 6; i++){
		printf("%d ", a[i]);
	}
	return 0;
}
